### PR TITLE
feat(confectionery): implement purchase endpoint, public access, DTO, frontend integration and success modal

### LINKED ISSUE #32: Confectionery: Side Panel Display for Selected Products (No Classic Cart)
## 🎯 Objective
Display a dynamic side section within the Confectionery page that shows selected products and their total cost, without using a traditional shopping cart model. This section should only appear when users are selecting food items and should behave differently depending on whether tickets were purchased first.

## 📋 Tasks
- [ ] Fetch confectionery products from backend API connected to the database
- [ ] Create side-panel layout for displaying selected products and prices
- [ ] Implement logic for adding and removing items from the side panel
- [ ] Show item quantity and subtotal in real-time
- [ ] Display a contextual notice if tickets were already purchased
- [ ] Ensure section is only visible in the Confectionery module

## 🔧 Technical Requirements
- Consume products from the backend via a RESTful API
- Use local storage or in-memory state (e.g., React Context, Zustand, etc.) to manage selections
- Apply smooth transition or slide-in animations
- Real-time price updates as products are selected
- Clear separation of UI state if ticket purchase happened before accessing Confitería

## ✅ Acceptance Criteria
- Products are successfully retrieved from the backend and rendered in the UI
- Users can clearly see their selected products and total in the side panel
- The interface does not resemble a typical "shopping cart"
- If accessed after buying tickets, the side panel adapts to indicate linked tickets
- No visual clutter or interference with other modules

### PR DESCRIPTION
**Pull request by Camilo-93**
## 📦 Description

This PR integrates **PayPal payment flow** for the confectionery module, including:
- Automatic currency conversion from **PEN to USD**
- Improved **stock validation**
- Prevention of **duplicate payments on reload**

It also enhances the **frontend user experience** by:
- Displaying **clear modals** (login, summary, success)
- Showing **unified total** (PEN + USD)
- Cleaning up the codebase (removed legacy functions like `handleBuy`)
- Commenting/documenting all logic in **English** for clarity

Closes: #32 

---

## ✅ Checklist

- [x] Code compiles and runs correctly
- [x] All existing and new tests pass
- [x] Follows coding conventions/style
- [x] No unnecessary `console.log`s or commented code
- [x] Proper error handling implemented
- [x] Responsive design (if applicable)
- [x] Connected to backend (if required)

---

## 🧪 How to Test

1. Go to the **Confitería** section in the app
2. Select one or more products and click **Finalize**
3. Test the **PayPal payment flow**, including:
   - Redirect to PayPal
   - Return to the app
   - Payment confirmation
4. Verify:
   - Totals are shown in **S/. (PEN)** and **USD**
   - Reloading after payment does **not** reprocess the transaction
   - Trying to buy more than available stock is blocked
   - All **modals display and behave correctly**

---

## 💬 Additional Notes

- PayPal integration now **requires all payments in USD**
- Stock is validated **before and after** payment to prevent overselling
- All logic and DTOs are **fully documented in English**
- Removed unused/legacy code for better maintainability

**Next steps**:
- Add unit tests for edge cases (e.g., PayPal cancellation, failure)
- Improve user feedback on backend/API errors

### CODE REVIEW

File: backend/src/main/java/com/cinerama/backend/controller/ConfectioneryPurchaseController.java [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+                    .orElseThrow(() -> new RuntimeException("Product not found: " + item.getProductId()));
+                if (product.getStock() < item.getQuantity()) {
+                    return ResponseEntity.badRequest().body("Insufficient stock for: " + product.getName());
+                }
+                totalAmountPEN += product.getPrice() * item.getQuantity();
+                product.setStock(product.getStock() - item.getQuantity());
+                productRepository.save(product);
+            }
+
+            // Convert total to USD (PayPal only accepts USD)
```
Comment (BruGeth) [inline backend/src/main/java/com/cinerama/backend/controller/ConfectioneryPurchaseController.java]: This could be simplified by calling convertSolesToDollars(amountInSoles) instead.

File: backend/src/main/java/com/cinerama/backend/controller/ConfectioneryPurchaseController.java [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+            double exchangeRate = 0.2818; // 1 sol ≈ 0.2818 USD
+            double totalAmountUSD = Math.round(totalAmountPEN * exchangeRate * 100.0) / 100.0;
+
+            // Create a paid order in USD
+            Order order = new Order();
+            order.setStatus("PAGADA");
```
Comment (BruGeth) [inline backend/src/main/java/com/cinerama/backend/controller/ConfectioneryPurchaseController.java]: Consider translating the order status values to Spanish and using lowercase (e.g., "PAGADA", "PENDIENTE") for consistency with localized responses.

File: backend/src/main/java/com/cinerama/backend/controller/ConfectioneryPurchaseController.java [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+            double totalAmountPEN = 0.0;
+            List<CartItem> cartItems = new ArrayList<>();
+            for (ConfectioneryPurchaseItemRequest item : request.getItems()) {
+                ConfectioneryProduct product = productRepository.findById(item.getProductId())
+                    .orElseThrow(() -> new RuntimeException("Product not found: " + item.getProductId()));
+                if (product.getStock() < item.getQuantity()) {
+                    return ResponseEntity.badRequest().body("Insufficient stock for: " + product.getName());
+                }
+                totalAmountPEN += product.getPrice() * item.getQuantity();
+                CartItem cartItem = new CartItem();
```
Comment (BruGeth) [inline backend/src/main/java/com/cinerama/backend/controller/ConfectioneryPurchaseController.java]: Also, ensure that the buyer's email is included in payer_email for better tracking and auditability.

File: frontend/src/pages/Confectionery.jsx [hunk 4]
```diff
@@ -1,95 +1,413 @@
...
+/* ===========================
+Cinerama Confectionery Page - React Component
+===========================
+- Handles category/product fetching, selection, and order summary
+- Integrates PayPal payment flow with currency conversion (PEN to USD)
+- Prevents duplicate payment processing on reload
+- Shows modals for login, summary, and purchase success
+- All logic and UI changes are clearly commented
+===========================*/
+import React, { useState, useEffect } from 'react';
```
Comment (BruGeth) [inline frontend/src/pages/Confectionery.jsx]: Check if all imported modules are being used—removing unused imports helps improve readability.

**Review by BruGeth — CHANGES_REQUESTED** [review]
You're doing great—solid progress so far. Please review and revise the code comments for clarity and consistency. Once updated, feel free to request a new review.

File: backend/src/main/java/com/cinerama/backend/controller/ConfectioneryPurchaseController.java [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+            String payerName = null;
+            if (SecurityContextHolder.getContext().getAuthentication() != null) {
+                payerEmail = SecurityContextHolder.getContext().getAuthentication().getName();
+            }
+            // Use the buyer name from the request, or default if not provided
+            payerName = request.getBuyerName() != null ? request.getBuyerName() : "Usuario Cinerama";
+
+            // Create a paid order in USD and associate the confectionery items
+            Order order = new Order();
+            order.setStatus("pagada");
```
Comment (BruGeth) [inline backend/src/main/java/com/cinerama/backend/controller/ConfectioneryPurchaseController.java]: The value assigned to the payment status should ideally come directly from the confirmed transaction flow to ensure accuracy and traceability.

File: backend/src/main/java/com/cinerama/backend/controller/ConfectioneryPurchaseController.java [hunk 6]
```diff
@@ -0,0 +1,346 @@
...
+            order.setTimestamp(LocalDateTime.now());
+            order.setCurrency("USD");
```
Comment (BruGeth) [inline backend/src/main/java/com/cinerama/backend/controller/ConfectioneryPurchaseController.java]: The currency should be derived from the transaction source to ensure reliable data representation, especially when dealing with multi-currency support.

File: backend/src/main/java/com/cinerama/backend/controller/ConfectioneryPurchaseController.java [hunk 7]
```diff
@@ -0,0 +1,346 @@
...
+                        .build();
+                orderItems.add(orderItem);
+            }
+
+            // Convert total to USD for PayPal using shared service method
+            double totalAmountUSD = paymentService.convertSolesToDollars(totalAmountPEN);
+
+            // Create a local pending order in USD and associate the confectionery items
+            Order order = new Order();
+            order.setStatus("pendiente");
```
Comment (BruGeth) [inline backend/src/main/java/com/cinerama/backend/controller/ConfectioneryPurchaseController.java]: The value assigned to the payment status should ideally come directly from the confirmed transaction flow to ensure accuracy and traceability.

File: backend/src/main/java/com/cinerama/backend/controller/ConfectioneryPurchaseController.java [hunk 8]
```diff
@@ -0,0 +1,346 @@
...
+            order.setTimestamp(LocalDateTime.now());
+            order.setCurrency("USD");
```
Comment (BruGeth) [inline backend/src/main/java/com/cinerama/backend/controller/ConfectioneryPurchaseController.java]: The currency should be derived from the transaction source to ensure reliable data representation, especially when dealing with multi-currency support.

File: backend/src/main/java/com/cinerama/backend/controller/ConfectioneryPurchaseController.java [hunk 9]
```diff
@@ -0,0 +1,346 @@
...
+                        "message", "Purchase was already processed",
+                        "paymentId", paymentId,
+                        "status", "already_completed"));
+            }
+
+            // If not paid, attempt to capture the payment
+            ResponseEntity<?> captureResponse = paymentService.capturePayment(paymentId, payerId);
+            if (captureResponse.getStatusCode().is2xxSuccessful()) {
+                // Update local order status to paid
+                order.setStatus("pagada");
```
Comment (BruGeth) [inline backend/src/main/java/com/cinerama/backend/controller/ConfectioneryPurchaseController.java]: The value assigned to the payment status should ideally come directly from the confirmed transaction flow to ensure accuracy and traceability.

File: frontend/package-lock.json
Comment (BruGeth) [inline frontend/package-lock.json:1]: PDF generation might be better handled on the backend for consistency and centralized control.

File: frontend/package.json
Comment (BruGeth) [inline frontend/package.json:1]: PDF generation might be better handled on the backend for consistency and centralized control.

File: frontend/src/pages/Confectionery.jsx
Comment (BruGeth) [inline frontend/src/pages/Confectionery.jsx:1]: PDF generation might be better handled on the backend for consistency and centralized control.

**Review by BruGeth — CHANGES_REQUESTED** [review]
Review completed. Once the proposed changes are applied, this can be considered ready for merge.

**Review by BruGeth — DISMISSED** [review]
LGTM!!!

**Review by BruGeth — APPROVED** [review]
Now merge.
